raiden/tests/integration/long_running/test_integration_events.py

```diff
@@ -522,24 +529,36 @@ def test_secret_revealed(raiden_chain, deposit, settle_timeout, token_addresses)
         secret,
     )
 
-    gevent.sleep(.1)  # wait for the messages
+    with gevent.Timeout(10):
+        wait_for_state_change(
+            app2.raiden,
+            ReceiveSecretReveal,
+            {'secrethash': secrethash},
+            retry_interval,
+        )
 
-    # The secret hasn't been revealed yet
     channel_state2_1 = get_channelstate(app2, app1, token_network_identifier)
-    assert len(channel_state2_1.our_state.secrethashes_to_lockedlocks) == 1
-
-    channel.register_offchain_secret(channel_state2_1, secret, secrethash)
-
-    # Close the channel
-    # This needs to register the secrets on chain
-    netting_channel_proxy = app2.raiden.chain.payment_channel(
-        token_network_identifier,
-        channel_state2_1.identifier,
-    )
-    netting_channel_proxy.channel_close(
-        registry_address,
-        channel_state2_1.partner_state.balance_proof,
-    )
+    pending_lock = channel_state2_1.partner_state.secrethashes_to_unlockedlocks.get(secrethash)
+    msg = "The lock must be registered in unlocked locks since the secret is known"
+    assert pending_lock is not None, msg
+
+    # The channels are out-of-sync. app1 has sent the unlock, however we are
+    # intercepting it and app2 has not received the updated balance proof
+    #
+    # assert_synced_channel_state(
```

Comment:
I suppose the commented-out assert was just to check if the channels are indeed out of sync? Shouldn't it be removed before merge?